plugin_tests/assetstore_test.py

```diff
@@ -0,0 +1,633 @@
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+##############################################################################
+
+import json
+import os
+import six
+from six.moves import urllib
+
+from girder import config
+from girder.constants import AssetstoreType, SettingKey
+from girder.models.model_base import GirderException
+from girder.utility import assetstore_utilities, progress
+from tests import base
+
+# boiler plate to start and stop the server
+
+os.environ['GIRDER_PORT'] = os.environ.get('GIRDER_TEST_PORT', '20201')
+config.loadConfig()  # Must reload config to pickup correct port
+
+
+def setUpModule():
+    base.enabledPlugins.append('database_assetstore')
+    base.startServer(False)
+
+
+def tearDownModule():
+    base.stopServer()
+
+
+class AssetstoreTest(base.TestCase):
+    def setUp(self):
+        base.TestCase.setUp(self)
+
+        users = ({
+            'email': 'good@email.com',
+            'login': 'goodlogin',
+            'firstName': 'First',
+            'lastName': 'Last',
+            'password': 'goodpassword'
+        }, {
+            'email': 'regularuser@email.com',
+            'login': 'regularuser',
+            'firstName': 'First',
+            'lastName': 'Last',
+            'password': 'goodpassword'
+        })
+        self.admin, self.user = [
+            self.model('user').createUser(**user) for user in users]
+        folders = self.model('folder').childFolders(
+            self.admin, 'user', user=self.admin)
+        for folder in folders:
+            if folder['name'] == 'Public':
+                self.publicFolder = folder
+            elif folder['name'] == 'Private':
+                self.privateFolder = folder
+        self.item1 = self.model('item').createItem(
+            'item1', creator=self.admin, folder=self.publicFolder)
+        self.item2 = self.model('item').createItem(
+            'item2', creator=self.admin, folder=self.publicFolder)
+        # Define locations for the default assetstores
+        self.dbParams = {
+            'type': AssetstoreType.DATABASE,
+            'name': 'Test Assetstore',
+            'dbtype': 'sqlalchemy_postgres',
+            'dburi': os.environ.get(
+                'GIRDER_DATABASE_ASSETSTORE_POSTGRES_DB',
+                'postgresql://postgres@127.0.0.1/sampledb'),
+        }
+        self.dbParams2 = {
+            'type': AssetstoreType.DATABASE,
+            'name': 'All Mongo Assetstores',
+            'dburi': '/'.join(os.environ.get('GIRDER_TEST_DB').split('/')[:-1])
+        }
+        self.dbParams3 = {
+            'type': AssetstoreType.DATABASE,
+            'name': 'One Mongo Assetstores',
+            'dburi': os.environ.get('GIRDER_TEST_DB')
+        }
+
+    def testAssetstoreCreate(self):
+        # Make sure admin access required
+        resp = self.request(path='/assetstore', method='POST',
+                            params=self.dbParams)
```

Comment:
A better assertion would be to request this as `self.user` and then assert 403 status.